observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerService.java

```diff
@@ -68,4 +63,52 @@ private void registerJmxMetrics() {
     public void close() {
         micrometerRegistry.close();
     }
+
+
+    void setMicrometerConfig(WildFlyMicrometerConfig micrometerConfig) {
+        this.micrometerConfig = micrometerConfig;
+    }
+
+    void setModelControllerClient(LocalModelControllerClient modelControllerClient) {
+        this.modelControllerClient = modelControllerClient;
+    }
+
+    void setProcessStateNotifier(ProcessStateNotifier processStateNotifier) {
+        this.processStateNotifier = processStateNotifier;
+    }
+
+    void setMicrometerRegistry(WildFlyCompositeRegistry micrometerRegistry) {
+        this.micrometerRegistry = micrometerRegistry;
+    }
+
+
+    public static class Builder {
+        private final MicrometerService service = new MicrometerService();
+
+        public Builder micrometerConfig(WildFlyMicrometerConfig micrometerConfig) {
+            service.setMicrometerConfig(micrometerConfig);
```

Comment:
Why is the service instance cached and setters used to configure that? 
Builders normally temporarily store the fields being set in the builder, and then call the constructor of whatever is being built in the build() method (in which case all the MicrometerService fields can be final again)
I don't see anything so heavy in the construction of MicrometerService warranting this strange approach :-)

Reply:
Yeah, I dunno. That is weird. I've modified that to be a proper builder.